src/elliptic_curve.rs

```diff
@@ -22,6 +22,22 @@ impl<F: Field + Clone + PartialEq> EllipticCurve<F> {
             weierstrass_coefficients: coeffs,
         }
     }
+    // Check that point is on the curve
+    pub fn is_on_curve(self, P: &ECPoint<F>) -> Result<ECPoint<F>, ErrorKind> {
+        let valid = false;
+        match (P) {
+            (ECPoint::PointAtInfinity) => Ok(P),
+            (ECPoint::AffinePoint(x_p,y_p)) => {
```

Comment:
```suggestion
        match P {
            ECPoint::PointAtInfinity => Ok(P),
            ECPoint::AffinePoint(x_p,y_p) => {
```
Parenthèses qui font des warnings